tests/e2e_logging.py

```diff
@@ -1630,23 +1634,36 @@ def test_committed_index(network, args):
 
     _, log_id = network.txs.get_log_id(txid)
 
-    r = network.txs.request(log_id, priv=True, url_suffix="committed")
+    remaining_retries = 10
+    while remaining_retries > 0:
+        r = network.txs.request(log_id, priv=True, url_suffix="committed")
+        if r.status_code == http.HTTPStatus.OK.value:
+            break
+
+        current_tx_id = TxID.from_str(
+            re.search(
+                r"Current Tx ID: ([0-9]+.[0-9]+)",
+                r.body.json()["error"]["message"],
+            ).group(1)
+        )
+
+        if current_tx_id >= txid:
+            break
+
+        remaining_retries -= 1
+
     assert r.status_code == http.HTTPStatus.OK.value, r.status_code
     assert r.body.json() == {"msg": f"Private message at idx {log_id} [0]"}
 
     network.txs.delete(log_id, priv=True)
 
     r = network.txs.request(log_id, priv=True)
     assert r.status_code == http.HTTPStatus.BAD_REQUEST.value, r.status_code
-    assert r.body.json() == {
-        "error": {"code": "ResourceNotFound", "message": f"No such record: {log_id}."}
-    }
+    assert r.body.json()["error"]["message"].startswith(f"No such record: {log_id}")
```

Comment:
Why are we weakening the assertion here?

Reply:
We only need to because we have the TxID in the message so once I do that in a better way this can also be fixed

Reply:
Change now lives in #4803